opendg/_io/tgb.py

```
@@ -0,0 +1,53 @@
+from typing import Any, List
+
+import torch
+from tgb.linkproppred.dataset import LinkPropPredDataset
+from tgb.nodeproppred.dataset import NodePropPredDataset
+
+from opendg.events import EdgeEvent, Event
+
+
+def read_tgb(
+    name: str,
+    split: str = 'all',  # Options: 'train', 'valid', 'test', 'all'
+    **kwargs: Any,
+) -> List[Event]:
+    # TODO: Node Events not supported
+    if name.startswith('tgbl-'):
+        dataset = LinkPropPredDataset(name=name, **kwargs)
+    elif name.startswith('tgbn-'):
+        dataset = NodePropPredDataset(name=name, **kwargs)
```

Do we support this right now? I thought not. If it works, then we can keep it

---

So as it stands right now, the `NodeEvents` are ignored but the `EdgeEvents` would be loaded. But also because of that the dataset is loaded incompletely. We should create a separate issue and PR to merge for the `NodePropPred` datasets. Maybe throw an error for now. not implemented error